### PR TITLE
feat: Go Plus Security plugin

### PR DESCRIPTION
**Pull request by Lanttcat**
## Description

Closes #MF-167

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/shared/index.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
 export * from './messages'
 export * from './flags'
 export { InMemoryStorages, PersistentStorages } from './kv-storage'
-export * from './helpers/download'
```
Comment (Jack-Works) [inline packages/mask/shared/index.ts]: looks like you're containing changes from other PRs.

Comment (Lanttcat) [reply packages/mask/shared/index.ts]: It depend pr: https://github.com/DimensionDev/Maskbook/pull/5939 .

**Comment by orange187** [comment]
Checked

File: packages/plugins/GoPlusSecurity/src/SNSAdaptor/CheckSecurityDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+export function CheckSecurityDialog(props: BuyTokenDialogProps) {
+    const t = useI18N()
+    const classes = useStylesExtends(useStyles(), props)
+    const { open, onClose } = props
+
+    const [{ value, loading: searching, error }, onSearch] = useAsyncFn(async (chainId: string, content: string) => {
+        const values = await GoPlusLabs.getTokenSecurity(chainId, [content.trim()])
+        if (!Object.keys(values ?? {}).length) throw new Error()
```
Comment (guanbinrui) [inline packages/plugins/GoPlusSecurity/src/SNSAdaptor/CheckSecurityDialog.tsx]: Leave an error message.

File: packages/plugins/GoPlusSecurity/src/SNSAdaptor/components/SearchBox.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        borderColor: theme.palette.divider,
+        borderRadius: 12,
+    },
+}))
+
+interface SearchBoxProps {
+    onSearch(chainId: string, content: string): void
+}
+
+const DEFAULT_SEARCH_CHAIN = '1'
```
Comment (guanbinrui) [inline packages/plugins/GoPlusSecurity/src/SNSAdaptor/components/SearchBox.tsx]:
```suggestion
const DEFAULT_SEARCH_CHAIN = ChainId.Mainnet.toString()
```

File: packages/plugins/GoPlusSecurity/src/SNSAdaptor/components/SearchBox.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+function getChainName(chain?: SecurityAPI.SupportedChain) {
+    if (!chain) return getChainDetailed(ChainId.Mainnet)?.chain
+    if (chain.id === ChainId.BSC.toString()) return getChainDetailed(ChainId.BSC)?.shortName.toUpperCase() ?? chain.name
```
Comment (guanbinrui) [inline packages/plugins/GoPlusSecurity/src/SNSAdaptor/components/SearchBox.tsx:47]:
```suggestion
    if (chain.id === ChainId.BSC) return getChainDetailed(ChainId.BSC)?.shortName.toUpperCase() ?? chain.name
```

File: packages/plugins/GoPlusSecurity/src/SNSAdaptor/CheckSecurityDialog.tsx (hunk 2 above)
Comment (Lanttcat) [reply packages/plugins/GoPlusSecurity/src/SNSAdaptor/CheckSecurityDialog.tsx]:
```
{error && !searching && <NotFound />}
```
I hope throw error to match error case.

Comment (guanbinrui) [reply packages/plugins/GoPlusSecurity/src/SNSAdaptor/CheckSecurityDialog.tsx]: You threw an error w/o error message.

File: packages/plugins/GoPlusSecurity/src/SNSAdaptor/components/SearchBox.tsx (hunk 3 above)
Comment (Lanttcat) [reply packages/plugins/GoPlusSecurity/src/SNSAdaptor/components/SearchBox.tsx]: I converted the chain id type in response from string to ChainId. https://github.com/DimensionDev/Maskbook/pull/5873/commits/6ca4e68f7c12d1008b274ee1f981ef465c780c7b

**Review by guanbinrui — DISMISSED** [review]

**Pull request merged**